Unable to start Quarkus with Camel FHIR

The project generated from https://code.quarkus.io with Camel FHIR dependency does not start.
Steps to reproduce:
- Goto https://code.quarkus.io/?e=org.apache.camel.quarkus%3Acamel-quarkus-fhir&extension-search=fhir
- Dowload the project
- Unzip and run `./mvnw quarkus:dev`
- It will crash with this exception:
```
2021-04-03 17:10:12,697 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.ChainBuildException: Cycle detected:
		   org.apache.camel.quarkus.component.fhir.deployment.r5.FhirR5Processor#recordContext produced class io.quarkus.arc.deployment.SyntheticBeanBuildItem
		to io.quarkus.arc.deployment.SyntheticBeansProcessor#initRegular produced class io.quarkus.arc.deployment.BeanRegistrationPhaseBuildItem$BeanConfiguratorBuildItem
		to io.quarkus.arc.deployment.ArcProcessor#registerSyntheticObservers produced class io.quarkus.arc.deployment.ObserverRegistrationPhaseBuildItem
		to io.quarkus.arc.deployment.ArcProcessor#validate produced class io.quarkus.arc.deployment.ValidationPhaseBuildItem
		to io.quarkus.arc.deployment.ObserverValidationProcessor#validateApplicationObserver produced class io.quarkus.arc.deployment.ValidationPhaseBuildItem$ValidationErrorBuildItem
		to io.quarkus.arc.deployment.ArcProcessor#generateResources produced class io.quarkus.arc.deployment.BeanContainerBuildItem
		to org.apache.camel.quarkus.component.fhir.deployment.r5.FhirR5Processor#recordContext
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:396)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:269)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:66)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:79)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:378)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:56)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:127)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:84)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:144)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: io.quarkus.builder.ChainBuildException: Cycle detected:
		   org.apache.camel.quarkus.component.fhir.deployment.r5.FhirR5Processor#recordContext produced class io.quarkus.arc.deployment.SyntheticBeanBuildItem
		to io.quarkus.arc.deployment.SyntheticBeansProcessor#initRegular produced class io.quarkus.arc.deployment.BeanRegistrationPhaseBuildItem$BeanConfiguratorBuildItem
		to io.quarkus.arc.deployment.ArcProcessor#registerSyntheticObservers produced class io.quarkus.arc.deployment.ObserverRegistrationPhaseBuildItem
		to io.quarkus.arc.deployment.ArcProcessor#validate produced class io.quarkus.arc.deployment.ValidationPhaseBuildItem
		to io.quarkus.arc.deployment.ObserverValidationProcessor#validateApplicationObserver produced class io.quarkus.arc.deployment.ValidationPhaseBuildItem$ValidationErrorBuildItem
		to io.quarkus.arc.deployment.ArcProcessor#generateResources produced class io.quarkus.arc.deployment.BeanContainerBuildItem
		to org.apache.camel.quarkus.component.fhir.deployment.r5.FhirR5Processor#recordContext
	at io.quarkus.builder.BuildChainBuilder.cycleCheck(BuildChainBuilder.java:362)
	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:335)
	at io.quarkus.builder.BuildChainBuilder.cycleCheck(BuildChainBuilder.java:366)
	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:335)
	at io.quarkus.builder.BuildChainBuilder.cycleCheck(BuildChainBuilder.java:366)
	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:335)
	at io.quarkus.builder.BuildChainBuilder.cycleCheck(BuildChainBuilder.java:366)
	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:335)
	at io.quarkus.builder.BuildChainBuilder.cycleCheck(BuildChainBuilder.java:366)
	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:335)
	at io.quarkus.builder.BuildChainBuilder.cycleCheck(BuildChainBuilder.java:366)
	at io.quarkus.builder.BuildChainBuilder.cycleCheckProduce(BuildChainBuilder.java:335)
	at io.quarkus.builder.BuildChainBuilder.cycleCheck(BuildChainBuilder.java:366)
	at io.quarkus.builder.BuildChainBuilder.cycleCheck(BuildChainBuilder.java:326)
	at io.quarkus.builder.BuildChainBuilder.build(BuildChainBuilder.java:236)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:137)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:394)
	... 9 more
```

--- a/extensions/fhir/deployment/src/main/java/org/apache/camel/quarkus/component/fhir/deployment/r5/FhirR5Processor.java
+++ b/extensions/fhir/deployment/src/main/java/org/apache/camel/quarkus/component/fhir/deployment/r5/FhirR5Processor.java
@@ -22,7 +22,6 @@ import java.util.Set;
 import javax.inject.Singleton;
 
 import ca.uhn.fhir.context.FhirContext;
-import io.quarkus.arc.deployment.BeanContainerBuildItem;
 import io.quarkus.arc.deployment.SyntheticBeanBuildItem;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
@@ -48,8 +47,7 @@ public class FhirR5Processor {
 
     @BuildStep(onlyIf = FhirFlags.R5Enabled.class)
     @Record(ExecutionTime.STATIC_INIT)
-    SyntheticBeanBuildItem recordContext(FhirContextRecorder fhirContextRecorder, BeanContainerBuildItem beanContainer,
-            R5PropertiesBuildItem propertiesBuildItem) {
+    SyntheticBeanBuildItem recordContext(FhirContextRecorder fhirContextRecorder, R5PropertiesBuildItem propertiesBuildItem) {
         return SyntheticBeanBuildItem.configure(FhirContext.class)
                 .scope(Singleton.class)
                 .named("R5")

